openapi_to_fastapi/cli.py

```diff
@@ -0,0 +1,102 @@
+import importlib.util
+import logging
+import sys
+import traceback
+import uuid
+from contextlib import contextmanager
+from pathlib import Path
+from typing import List
+
+import click
+import coloredlogs
+from fastapi import APIRouter
+
+from openapi_to_fastapi.validator.core import BaseValidator, DefaultValidator
+
+from .routes import validate_spec_and_create_routes
+from .validator import ihan_standards
+
+logger = logging.getLogger("openapi_to_fastapi_cli")
+coloredlogs.install(logger=logger, fmt="%(message)s")
+
+
+def print_dashes(char="=", length=79):
+    logger.info(char * length)
+
+
+@contextmanager
+def header():
+    print_dashes()
+    yield
+    print_dashes()
+
+
+@click.command()
+@click.option(
+    "--path", "-p", required=True, help="Path to directory with OpenAPI specs"
+)
+@click.option("--validator", "-v", multiple=True, help="Class name of extra validator")
+@click.option(
+    "--module",
+    "-m",
+    multiple=True,
+    help="Module with extra validator class definitions",
+)
+def cli_validate_specs(path: str, validator, module):
+    sys.exit(validate_specs(Path(path), module, validator))
+
+
+def validate_specs(path: Path, modules: List[str], extra_validators: List[str]) -> int:
+    validators = _load_validators(extra_validators, modules)
+    with header():
+        logger.info(f"OpenAPI specs root path: {path}")
+        logger.info("Validators: %s", ", ".join([v.__name__ for v in validators]))
+
+    passed, failed = 0, 0
+    for spec_path in path.glob("**/*.json"):
+        logger.info(f"File: {spec_path}")
+        try:
+            validate_spec_and_create_routes(APIRouter(), spec_path, validators)
+        except Exception:
+            logger.error("\n%s", traceback.format_exc())
+            logger.error("[FAILED]")
+            failed += 1
+        else:
+            logger.info("[PASSED]")
+            passed += 1
+        print_dashes("-")
+
+    with header():
+        log = logger.error if failed else logger.info
+        log("Summary:")
+        log(f"Total : {passed+failed}")
+        log(f"Passed: {passed}")
+        log(f"Failed: {failed}")
+    return failed
```

Comment:
This is a bad idea, typically posix expects this to be in the range of 0-126, and can be at most 255. Yet it is easily possible someone has more than 255 files. Just use exit code 1 if there were any errors.

Reply:
Typically exit codes are used to discern different types of errors when it makes sense, not numbers of errors.